rllib/core/learner/learner.py

```diff
@@ -865,33 +1022,29 @@ def additional_update_for_module(
 
         Args:
             module_id: The id of the module to update.
+            hps: The LearnerHyperparameters specific to the given `module_id`.
             timestep: The current global timestep (to be used with schedulers).
             **kwargs: Keyword arguments to use for the additional update.
 
         Returns:
             A dictionary of results from the update
         """
-        return {}
+        results = {}
 
-    @OverrideToImplementCustomLogic
-    def postprocess_gradients(
-        self,
-        gradients_dict: Mapping[str, Any],
-    ) -> Mapping[str, Any]:
-        """Applies potential postprocessing operations on the gradients.
-
-        This method is called after gradients have been computed, and modifies them
-        before they are applied to the respective module(s).
-        This includes grad clipping by value, norm, or global-norm, or other
-        algorithm specific gradient postprocessing steps.
+        # Handle lr-scheduling updates and apply new learning rates to the optimizers.
+        if not self._user_configured_optimizers:
```

Comment:
We only auto-manage lr-scheduling iff user has NOT overridden `configure_optimizer_for_module` method.

Reply:
An alternative suggestion:
- Leave lr-updates inside additional-update method (it's NOT a gradient based, train batch-dependent "update" of the model)
- `Update` may or may not receive an additional timestep arg, but I don't think it's necessary as users should NOT override `update` anyways.
- Everything that the user wants to update every iteration anyways (and where there is no question that is should be updated every iteration), such as lr, entropy_coeff, kl, etc.. should NOT require a separate boolean arg in `addition_updates_for_module`'s signature. That seems overkill. If we argue against this, then we shouldn't even have a single `addition_update_for_module` method, but split it up into algorithm specific calls (similar to PPO's `update_kl` Policy mix-in methods :| ).
- For things like target net update, though, we should have it as target nets don't get updated every iteration.

Reply:
ok so this will be the TLDR of our conversations (The concrete convention)

The update and additional_update are assumed to be called on every iteration.
Scheduler stuff will get updated on every iterations. 
Things that may need a different frequency will have a dedicated flag to control what gets updated.